src/IceRpc.Protobuf/MessageParserExtensions.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) ZeroC, Inc.
+
+using Google.Protobuf;
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Diagnostics;
+using System.IO.Pipelines;
+
+namespace IceRpc.Protobuf;
+
+/// <summary>Provides an extension method for <see cref="MessageParser{T}" />.</summary>
+public static class MessageParserExtensions
+{
+    /// <summary>Decodes a Protobuf length prefixed message from a <see cref="PipeReader" />.</summary>
+    /// <param name="parser">The <see cref="MessageParser{T}" /> used to parse the message data.</param>
+    /// <param name="reader">The <see cref="PipeReader" /> containing the Protobuf length prefixed message.</param>
+    /// <param name="maxMessageLength">The maximum allowed length.</param>
+    /// <param name="cancellationToken">A cancellation token that receives the cancellation requests.</param>
+    /// <returns>The decoded message object.</returns>
+    public static async ValueTask<T> DecodeFromLengthPrefixedMessageAsync<T>(
+        this MessageParser<T> parser,
+        PipeReader reader,
+        int maxMessageLength,
+        CancellationToken cancellationToken) where T : IMessage<T>
+    {
+        ReadResult readResult = await reader.ReadAtLeastAsync(5, cancellationToken).ConfigureAwait(false);
```

Comment:
Check readResult.IsCanceled.

This is similar to this code:
https://github.com/icerpc/icerpc-csharp/blob/ecbe38c0fd584046384b0589b9c94b278fb5ca6f/src/IceRpc.Slice/Internal/IncomingFrameExtensions.cs#L39